indico/web/client/js/react/components/principals/Search.jsx

```diff
@@ -431,14 +444,60 @@ const InnerUserSearch = searchFactory({
       return handleSubmitError(error);
     }
     const resultData = camelizeKeys(response.data);
-    resultData.results = resultData.users.map(({identifier, id, fullName, email, affiliation}) => ({
-      identifier,
-      type: PrincipalType.user,
-      userId: id,
-      name: fullName,
-      detail: affiliation ? `${email} (${affiliation})` : email,
-    }));
+    resultData.results = resultData.users.map(
+      ({identifier, id, fullName, email, affiliation, firstName, lastName}) => ({
+        identifier,
+        type: PrincipalType.user,
+        userId: id,
+        id,
+        name: fullName,
+        detail: affiliation ? `${email} (${affiliation})` : email,
+        firstName,
+        lastName,
+        existsInEvent: false,
+        email,
+        affiliation,
+      })
+    );
     delete resultData.users;
+    if (withEventPersons) {
+      let epResponse;
+      values.confId = eventId;
```

Comment:
Instead of modifying this I'd rather do something like this below:

```
        epResponse = await indicoAxios.get(eventPersonSearchURL({...values, confId: eventId}));
```

Assuming `favorites_first` isn't event used in this endpoint here, we could do the same for it above, so `values` really only contains the search criteria (in which case I'd rename it to something like `criteria`).